Assets/MixedRealityToolkit/Extensions/VectorExtensions.cs

```diff
@@ -162,6 +162,56 @@ public static bool IsValidVector(this Vector3 vector)
                    !float.IsInfinity(vector.x) && !float.IsInfinity(vector.y) && !float.IsInfinity(vector.z);
         }
 
+        /// <summary>
+        /// Calculates which side a point in world space is on of a plane
+        /// </summary>
+        /// <param name="initialDirection">The normal direction orthagonal to the plane. </param>
+        /// <param name="initialPosition">The point representing the plane's origin</param>
+        /// <returns>true when the compared point is on the other side of the plane</returns>
+        public static bool IsOtherSideOfPoint(this Vector3 vector, Vector3 initialDirection, Vector3 initialPosition)
+        {
+            Vector3 delta = vector - initialPosition;
+            float dot = Vector3.Dot(delta.normalized, initialDirection);
+
+            return (dot > 0) ? true : false;
+        }
+
+        /// <summary>
+        /// Projects a pointer's actual direction on the same plane as a previous point in space.
+        /// </summary>
+        /// <param name="initialPoint">The original point representing the plane to project <paramref name="newDir"/> on.</param>
+        /// <param name="origin">Where <paramref name="newDir"/> is originating from in world space.</param>
+        /// <param name="newDir">The direction to project</param>
+        /// <param name="axisConstraint">The (optional) axis to scale to</param>
+        /// <returns>><see cref="Vector3"/> representing the new point on the plane as the hit point in world space</returns>
+        public static Vector3 GetProjectedPoint(Vector3 initialPoint, Vector3 origin, Vector3 newDir, Vector3? directionalConstraint)
```

Comment:
This doesn't feel like a general utility to me. I actually don't understand what it's doing...perhaps move it back into scrollable collection for now? Or maybe you can help me understand what the code does.

Reply:
This is how I was finding a new point on the plane of the pointer's hit when scrolling with rays. Since you've taken a pass I'll go ahead and pull this out.